x/auth/types/credentials.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"fmt"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+)
+
+// NewAccountWithModuleCredential creates an account with an unclaimable module credential.
+func NewAccountWithModuleCredential(pubkey cryptotypes.PubKey) (*BaseAccount, error) {
```

Comment:
Don't we already have a method that creates a baseaccount with a public key?

Reply:
But this isn't specific to module credentials right?

Reply:
Sure let me rename.